support no-response option

arun1587 opened the pull request:
OptionID has to be made uint16 in order to support the no-response
option number i.e. 258.
Added Encoding and Decoding message tests for the no-response option.
Reference: https://tools.ietf.org/html/rfc7967

jkralik commented:
Thank you for contributing! Please could you:
- update file README.MD about this RFC
- create NoResponseWriter that satisfy interface -> it will writes only message with allowed code, otherwise it returns error.

jkralik commented on noresponsewriter.go:
```diff
@@ -0,0 +1,76 @@
+}
+
+type noResponseWriter struct {
+	*responseWriter
```
please replace `*responseWriter` with `responseWriter ResponseWriter`

jkralik commented on noresponsewriter.go:
```diff
@@ -0,0 +1,76 @@
+	noResponseValueMap map[uint32][]COAPCode
+}
+
+func newNoResponseWriter(w ResponseWriter) *noResponseWriter {
```
please use factory at line https://github.com/go-ocf/go-coap/blob/3638c5d5aefdb54c24ba9ec02dddf113bed6a49c/responsewriter.go#L64

jkralik commented on noresponsewriter.go:
```diff
@@ -0,0 +1,76 @@
+}
+
+func (w *noResponseWriter) SetCode(code COAPCode) {
+	w.code = &code
```
pls forward call to responseWriter `w.responseWriter.SetCode(code)`

arun1587 commented:
> Please add test for noResponseWriter, add update MR against new go-coap.
> Thx
I rebased on recent changes. added tests for decoding but can you give any reference for improving the coverage, especially the methods of `ResponseWriter`.

jkralik commented:
Create server via `RunLocalServerUDPWithHandler("udp", ":", false,  BlockWiseSzx16, func (w ResponseWriter, r *Request) { testNoResponseHandler(t, w, r }) `  - It returns server, addr where client need to connect and fin channel (when server you call shutdown then read fin chan).  And connect client, create req via New{Get/Post/...}Request, set option NoResponse and test with WriteMsg/ExchangeMsg.

arun1587 commented:
You could squash the commits before merge.,
Thanks.

jkralik commented:
Good works !
